.emacs.d/lisp/package/git-update.el

```diff
@@ -0,0 +1,79 @@
+;;; THIS FILE IS NOT LOADED AT STARTUP.
+;;; This file is autoloaded on the following symbols:
+;;;  kotct/update-dot-config
+;;;  kotct/update-user-config
+
+(require 'user-config-system)
+
+(package-initialize)
+
+(defvar kotct/directory "~/.emacs.d"
+  "The path to your instance of kotct/dot or one of it's subdirectories.")
+
+(defun kotct/git-repository-diff (directory)
+  "Fetches in DIRECTORY and returns the difference between the common ancestor of HEAD and origin/master in DIRECTORY."
+  (let ((default-directory directory))
+    (kotct/run-git "fetch" "origin" "master")
+    (kotct/run-git "diff" "HEAD...origin/master")))
```

Comment:
I am not sure if I mentioned this in my previous review, (certainly doesn't appear so) but I am not sure if this is what you want.

For reference, the docs for `git-diff(1)`:

>       git diff [<options>] <commit> <commit> [--] [<path>...]
>           This is to view the changes between two arbitrary <commit>.

>       git diff [<options>] <commit>..<commit> [--] [<path>...]
>           This is synonymous to the previous form. If <commit> on one side is
>           omitted, it will have the same effect as using HEAD instead.

>       git diff [<options>] <commit>...<commit> [--] [<path>...]
>           This form is to view the changes on the branch containing and up to
>           the second <commit>, starting at a common ancestor of both
>           <commit>. "git diff A...B" is equivalent to "git diff $(git
>           merge-base A B) B". You can omit any one of <commit>, which has the
>           same effect as using HEAD instead.

If I'm reading this right, `A...B` represents the symmetric difference between the two refs `A` and `B`, i.e. the commits present in either but not both of them. But `A..B` just works in one direction; it takes the commits in `B` that are not in `A`, so if `A` includes extra commits it still doesn't print them.

Since I think the usage of this `kotct/git-repository-diff` function (below) is just to see if there are any new changes on `origin/master`, I think the appropriate form is `..`. Otherwise, if local commits are made _ahead of_ `origin/master`, this would produce nonempty output.

Reply:
Nope, I'm wrong. What I literally just posted says:

>`git diff A...B` is equivalent to `git diff $(git merge-base A B) B`

(`git merge-base A B` finds the oldest common ancestor.)

So yeah this is 100% right. Resolving my comment.